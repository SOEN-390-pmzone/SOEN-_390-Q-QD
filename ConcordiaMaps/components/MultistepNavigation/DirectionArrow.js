import React from 'react';
import { View, StyleSheet } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

const DirectionArrow = () => (
  <View style={styles.container}>
    <Ionicons name="chevron-down" size={28} color="#912338" style={styles.icon} />
  </View>
);

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    paddingVertical: 8,
  },
  icon: {
    textShadowColor: 'rgba(0, 0, 0, 0.2)',
    textShadowOffset: { width: 1, height: 1 },
    textShadowRadius: 1,
  }
});

export default DirectionArrow;