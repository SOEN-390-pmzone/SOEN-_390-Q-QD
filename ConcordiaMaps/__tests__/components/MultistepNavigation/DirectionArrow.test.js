import React from "react";
import { render } from "@testing-library/react-native";
// import { Ionicons } from '@expo/vector-icons';
import DirectionArrow from "../../../components/MultistepNavigation/DirectionArrow"; // Adjust import path as needed

// Mock the Ionicons component
jest.mock("@expo/vector-icons", () => ({
  Ionicons: () => null, // Return a simple null component for testing
}));

describe("DirectionArrow Component", () => {
  it("renders correctly", () => {
    const { getByTestId } = render(<DirectionArrow />);

    const directionArrowContainer = getByTestId("direction-arrow-container");
    expect(directionArrowContainer).toBeTruthy();
  });

  it("matches snapshot", () => {
    const { toJSON } = render(<DirectionArrow />);

    expect(toJSON()).toMatchSnapshot();
  });
});
