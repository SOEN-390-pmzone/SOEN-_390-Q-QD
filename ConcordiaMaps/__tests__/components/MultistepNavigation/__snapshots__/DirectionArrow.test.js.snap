// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DirectionArrow Component matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "paddingVertical": 8,
    }
  }
  testID="direction-arrow-container"
/>
`;
